ssas/service/public/api.go

```diff
@@ -467,3 +467,44 @@ func token(w http.ResponseWriter, r *http.Request) {
 
 	render.JSON(w, r, m)
 }
+
+func introspect(w http.ResponseWriter, r *http.Request) {
+	clientID, secret, ok := r.BasicAuth()
+
+	if !ok {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	system, err := ssas.GetSystemByClientID(clientID)
+	if err != nil {
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	savedSecret, err := system.GetSecret()
+	if err != nil || !ssas.Hash(savedSecret).IsHashOf(secret) {
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	defer r.Body.Close()
+
+
+	var reqV map[string]string
+	if err = json.NewDecoder(r.Body).Decode(&reqV); err != nil {
+		http.Error(w, "invalid body", http.StatusBadRequest)
+		return
+	}
+	var answer = make(map[string]bool)
+	answer["active"] = true
+	if err = tokenValidity(reqV["token"], "AccessToken"); err != nil {
+		answer["active"] = false
+	}
+
```

Comment:
Should we add a "TODO" placeholder here to remind us to test whether the token has been revoked?

Reply:
I had assumed that the `tokenValidity()` function would do that. Is there any reason for it not to?

Reply:
That probably makes sense.  I'll alter my PR to put it in there.